lms/static/scripts/frontend_apps/utils/fetch.ts

```diff
@@ -108,6 +121,21 @@ export function useFetch<T = unknown>(
               mutate,
               retry: doFetch,
             });
+
+            // If an API call returns an unauthorized or forbidden response, it
+            // means the user should not have ended up here in the first place.
+            if (
+              reportToInitialLoading &&
+              error instanceof APIError &&
+              [401, 403].includes(error.status)
```

Comment:
Why not 404 or 5xx errors? All 4xx or 5xx errors seem equally fatal in this context.

Reply:
I wanted to focus on the authorization errors specifically, with the intention to later add some clear instructions on how the user can resolve it: "Go back to the assignment, then click Open Dashboard again".

I think in general, the errors we should report as fatal errors are those for which we have a clear list of steps to show the user, or we know exactly what's the only reason that could cause them, so we can display very specific information. Other errors could just be handled by individual components with a more generic error message.

But I'm happy reporting any non-success API call as a fatal error, and display concrete information when we can, or a more generic error when the reason is less clear.

Reply:
Hmm... it will make for a simpler API if all 4xx and 5xx errors are handled the same way. However if we have a more complex UI with multiple "high priority"/render-blocking fetches it could be undesirable to hide all of the content if only some parts fail to load. I think I would lean towards keeping the logic simpler, then making it more flexible when we have specific use cases.